contracts/moda-contracts/src/ModaRegistry.sol

```diff
@@ -3,22 +3,158 @@ pragma solidity 0.8.21;
 
 import "./interfaces/IModaRegistry.sol";
 import "./interfaces/IMembership.sol";
+import "@openzeppelin/contracts/access/extensions/AccessControlEnumerable.sol";
 
-contract ModaRegistry {
-    mapping(string => address) _catalogMemberships;
+contract ModaRegistry is IModaRegistry, AccessControlEnumerable {
+    using EnumerableSet for EnumerableSet.AddressSet;
 
+    /// State Variables
 
-    function isMember(string calldata catalogName, address user) external view returns (bool) {
-        address membership = _catalogMemberships[catalogName];
+    uint256 _catalogCount;
+    address payable _treasury;
+    uint256 _treasuryFee;
 
-        if (IMembership(membership).isMember(user)) {
+    mapping(string => Catalog) _catalogs;
```

Comment:
I think `catalogs` should be an array, not a mapping. There probably won't be many catalogs so a key name lookup is not necessary.